src/sourcemaps/tools/webpack.ts

```diff
@@ -47,21 +89,297 @@ export const configureWebPackPlugin: SourceMapUploadToolConfigurationFunction =
+      (await askForToolConfigPath('Webpack', 'webpack.config.js'));
 
-    // Intentially logging directly to console here so that the code can be copied/pasted directly
-    // eslint-disable-next-line no-console
-    console.log(getCodeSnippet(options));
+    const successfullyAdded = webpackConfigPath
+      ? await modifyWebpackConfig(webpackConfigPath, options)
+      : await createNewConfigFile(
+          path.join(process.cwd(), 'webpack.config.js'),
+          getCodeSnippet(options, false),
+          'More information about Webpack configs: https://vitejs.dev/config/',
+        );
 
-    await abortIfCancelled(
-      select({
-        message: 'Did you copy the snippet above?',
-        options: [{ label: 'Yes, continue!', value: true }],
-        initialValue: true,
-      }),
-    );
+    if (successfullyAdded) {
+      Sentry.setTag('ast-mod', 'success');
+    } else {
+      Sentry.setTag('ast-mod', 'fail');
+      await showCopyPasteInstructions(
+        path.basename(webpackConfigPath || 'webpack.config.js'),
+        getCodeSnippet(options, true),
+      );
+    }
 
     await addDotEnvSentryBuildPluginFile(options.authToken);
   };
+
+/**
+ * Modifies a webpack config file to enable source map generation and add the Sentry webpack plugin
+ * exported only for testing
+ */
+export async function modifyWebpackConfig(
+  webpackConfigPath: string,
+  options: SourceMapUploadToolConfigurationOptions,
+): Promise<boolean> {
+  try {
+    const webpackConfig = await fs.promises.readFile(webpackConfigPath, {
+      encoding: 'utf-8',
+    });
+
+    const prettyConfigFilename = chalk.cyan(path.basename(webpackConfigPath));
+
+    // no idea why recast returns any here, this is dumb :/
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    const program = recast.parse(webpackConfig.toString()).program as t.Program;
+
+    if (hasSentryContentCjs(program)) {
+      const shouldContinue = await abortIfCancelled(
+        clack.select({
+          message: `Seems like ${prettyConfigFilename} already contains Sentry-related code. Should the wizard modify it anyway?`,
+          options: [
+            {
+              label: 'Yes, add the Sentry Webpack plugin',
+              value: true,
+            },
+            {
+              label: 'No, show me instructions to manually add the plugin',
+              value: false,
+            },
+          ],
+          initialValue: true,
+        }),
+      );
+
+      if (!shouldContinue) {
+        Sentry.setTag('ast-mod-fail-reason', 'has-sentry-content');
+        return false;
+      }
+    }
+
+    const exportStmt = getCjsModuleExports(program);
+    if (!exportStmt) {
+      // We only care about CJS at the moment since it's probably the most widely used format for webpack configs.
+      clack.log.warn(
+        `Could not find module.exports = { /* config */ } in ${prettyConfigFilename}.
+This is fine, please follow the instructions below.`,
+      );
```

Comment:
l/m: I don't think we should show this log message (or any of the log messages that we couldn't find something in the AST) and I think we should just go for the fallback without saying anything. Will people care about these messages? Otherwise, they're just noise. Wdyt? (we should still do telemetry tho)

Reply:
Fair, I agree, it's a little too verbose here. I'll replace it with a `debug` call. Might still be useful to keep this around for future debugging.